### PR TITLE
doc: add example of how to filter based on node properties

### PR DESCRIPTION
**Pull request by larsmoa**
Also fixes an issue causing Cognite3dModel.showNodeByTreeIndex() not to work when applyToChildren was true.

### CODE REVIEW

File: documentation/docs/examples/cad-nodefiltering.mdx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<DemoWrapper
+  modelId={4715379429968321}
+  revisionId={5688854005909501}
+  name="Cognite3DViewerDemo"
+/>
+
+## Only show nodes that has a given node attribute value
+
+The available node properties varies a lot from model to model, and to use this information you will need to investigate
+the structure of the underlying models. For the purpose of this example, we the attributes of the nodes typically looks something like:
```
Comment (maksnester) [inline documentation/docs/examples/cad-nodefiltering.mdx:24]: >  For the purpose of this example, we the attributes of the nodes typically looks something like

Could you rephrase that part? I'm not sure what to suggest.

File: documentation/docs/examples/cad-nodefiltering.mdx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```json
+"PDMS": {
```
Comment (maksnester) [inline documentation/docs/examples/cad-nodefiltering.mdx:26]: That thing looks scary to me. Maybe to simplify it?

File: documentation/docs/examples/cad-nodefiltering.mdx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from the [Cognite SDK](https://www.npmjs.com/package/@cognite/sdk) which maps to the [List 3D Nodes API endpoint](https://docs.cognite.com/api/v1/#operation/get3DNodes).
+For this example we query for nodes that are within the 'EP' function and hide all other nodes.
+
+<LiveCodeSnippet>
+{`
+model.hideAllNodes();
+sdk.revisions3D.list3DNodes(model.modelId, model.revisionId, 
+  { 
+    limit: 1000, 
+    properties: { PDMS: { Function: 'EA' } }
```
Comment (maksnester) [inline documentation/docs/examples/cad-nodefiltering.mdx:50]: I suggest using something more readable to demonstrate the point that we can filter properties. Function "EA" doesn't make any sense to me personally, so it's just harder to read than some filtering by name/type/date or whatever more meaningful.

File: examples/src/pages/Migration.tsx [hunk 4]
```diff
@@ -68,6 +68,7 @@ export function Migration() {
       });
       (window as any).viewer = viewer;
 
+
```
Comment (maksnester) [inline examples/src/pages/Migration.tsx:71]:
```suggestion
```

File: viewer/src/public/migration/Cognite3DModel.ts [hunk 5]
```diff
@@ -487,8 +487,8 @@ export class Cognite3DModel extends THREE.Object3D implements CogniteModelBase {
    */
   async showNodeByTreeIndex(treeIndex: number, applyToChildren = false): Promise<number> {
     const treeIndices = await this.determineTreeIndices(treeIndex, applyToChildren);
-    this.hiddenNodes.delete(treeIndex);
-    this.cadNode.requestNodeUpdate([treeIndex]);
+    treeIndices.forEach(idx => this.hiddenNodes.delete(idx));
+    this.cadNode.requestNodeUpdate(treeIndices.asArray());
```
Comment (maksnester) [inline viewer/src/public/migration/Cognite3DModel.ts:491]: That's not related to that PR at all, but I think that conversion of treeIndices to an array should be called `toArray` I guess that name is more common to what we have in js with its toString toNumber toLowercase and other similarly named functions.

**Review by maksnester — CHANGES_REQUESTED** [review]

File: documentation/docs/examples/cad-nodefiltering.mdx (hunk 3 above)
Comment (larsmoa) [reply documentation/docs/examples/cad-nodefiltering.mdx:50]: I agree that this doesn't make much sense, but I haven't really been able to find good examples of filters I think make sense. The input model is not great when it comes to node metadata so finding good examples is hard.

File: viewer/src/public/migration/Cognite3DModel.ts (hunk 5 above)
Comment (larsmoa) [reply viewer/src/public/migration/Cognite3DModel.ts]: Agreed, fixed.

File: documentation/docs/examples/cad-highlighting.mdx [hunk 6]
```diff
@@ -106,7 +106,7 @@ viewer.on('click', event => {
         if (taggedEquipment) {
           // Highlight tagged equipment and all children
           model.selectNodeByTreeIndex(taggedEquipment.treeIndex, true);
-        }
+        }
```
Comment (maksnester) [inline documentation/docs/examples/cad-highlighting.mdx:109]:
```suggestion
        }
```

Will create a task on miro and try to handle linter for all packages, not only the viewer...

**Review by maksnester — APPROVED** [review]
LGTM. 

I removed auto-merge in case if you're on something still.
